src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs

```diff
@@ -345,74 +336,90 @@ private string GetTestHostPath(string runtimeConfigDevPath, string depsFilePath,
-                        try
-                        {
-                            testHostFullPath = Path.Combine(x.ToString(), testHostPath);
-                        }
-                        catch (ArgumentException)
+                        JObject context = (JObject)JToken.ReadFrom(reader);
+                        JObject runtimeOptions = (JObject)context.GetValue("runtimeOptions");
+                        JToken additionalProbingPaths = runtimeOptions.GetValue("additionalProbingPaths");
+                        foreach (var x in additionalProbingPaths)
                         {
-                            // https://github.com/Microsoft/vstest/issues/847
-                            // skip any invalid paths and continue checking the others
-                            continue;
-                        }
+                            EqtTrace.Verbose("DotnetTestHostmanager: Looking for path {0} in folder {1}", testHostPath, x.ToString());
+                            string testHostFullPath;
+                            try
+                            {
+                                testHostFullPath = Path.Combine(x.ToString(), testHostPath);
+                            }
+                            catch (ArgumentException)
+                            {
+                                // https://github.com/Microsoft/vstest/issues/847
+                                // skip any invalid paths and continue checking the others
+                                continue;
+                            }
 
-                        if (this.fileHelper.Exists(testHostFullPath))
-                        {
-                            return testHostFullPath;
+                            if (this.fileHelper.Exists(testHostFullPath))
+                            {
+                                return testHostFullPath;
+                            }
                         }
                     }
                 }
             }
+            else
+            {
+                errorMessage = string.Format(CultureInfo.CurrentCulture, Resources.UnableToFindDepsFile, depsFilePath);
+            }
 
-            // If we are here it means it couldnt resolve testhost.dll from nuget chache.
+            // If we are here it means it couldnt resolve testhost.dll from nuget cache.
             // Try resolving testhost from output directory of test project. This is required if user has published the test project
             // and is running tests in an isolated machine. A second scenario is self test: test platform unit tests take a project
             // dependency on testhost (instead of nuget dependency), this drops testhost to output path.
             testHostPath = Path.Combine(sourceDirectory, "testhost.dll");
             EqtTrace.Verbose("DotnetTestHostManager: Assume published test project, with test host path = {0}.", testHostPath);
 
+            if (!this.fileHelper.Exists(testHostPath))
```

Comment:
seems like we could simply replace the message above at line 216 with this new resource?

Reply:
We are caching the error above, and then throwing that.